libs/kotlinlib/src/mill/kotlinlib/ksp/KspModule.scala

```diff
@@ -360,6 +390,108 @@ trait KspModule extends KotlinModule { outer =>
 
   }
 
+  /**
+   * The classloader with [[ksp2InProgramToolsClasspath]], which includes the KSP 2 Worker
+   * and the user defined symbol processors via [[kotlinSymbolProcessors]]. This classloader is a
+   * parent of [[kotlinSymbolProcessorClassloader]] as classes are shared between the [[ksp2ToolsDeps]]
+   * and [[kotlinSymbolProcessors]] (symbol processors depend on the KSP API).
+   */
+  def ksp2WorkerClassloader: Worker[MillURLClassLoader] = Task.Worker {
```

Comment:
Does this really need to be a `MillURLClassLoader`? Why?

Reply:
I thought this was the type typically used. Should it just be URLClassloader?

Reply:
I guess `ClassLoader` would be enough. In the end, all we want is loading classes, and we don't care however this is managed. This gives more flexibility to our users.